src/cluster/cluster-unique-state-provider.js

```diff
@@ -9,43 +9,75 @@ SUPPORTED_ACTIONS[ C.ACTIONS.LOCK_RESPONSE ] = true;
 SUPPORTED_ACTIONS[ C.ACTIONS.LOCK_REQUEST ] = true;
 SUPPORTED_ACTIONS[ C.ACTIONS.LOCK_RELEASE ] = true;
 
-
 /**
- * Uses options
- *      uniqueTimeout
- *      leaderResponseTimeout
+ * The unique registry is responsible for maintaing a single source of truth
+ * within the cluster, used mainly for issuing cluster wide locks when an operation
+ * that stretches over multiple nodes are required.
+ *
+ * For example, distributed listening requires a leader to drive the nodes in sequence,
+ * so issuing a lock prevents multiple nodes from assuming the lead.
+ *
+ * Uses options:
+ *      lockTimeout
+ *      lockRequestTimeout
  *
- * Uses topics
- *      C.TOPIC.LEADER
- *      C.TOPIC.LEADER_PRIVATE_<serverName>
+ * Uses topics:
+ * 	    C.TOPIC.LEADER
+ * 	    C.TOPIC.LEADER_PRIVATE_<serverName>
  *
- * Uses action
- *      C.ACTIONS.LEADER_REQUEST
- *      C.ACTIONS.LEADER_VOTE
+ * Uses action:
+ * 		C.ACTIONS.LEADER_REQUEST
+ * 		C.ACTIONS.LEADER_VOTE
  */
-module.exports = class UniqueRegistry{
+module.exports = class UniqueRegistry {
+	/**
+	 * The unique registry is a singleton, and is only created once
+	 * within deepstream.io, and is injected into its dependencies
+	 * via the options object.
+	 *
+	 * @param  {Object} options         The options deepstream was created with
+	 * @param  {ClusterRegistry} clusterRegistry The cluster registry, used to get the cluster leader
+	 *
+	 * @constructor
+	 */
 	constructor( options, clusterRegistry ) {
 		this._options = options;
 		this._clusterRegistry = clusterRegistry;
 		this._locks = {};
-		this._lockTimeouts = {};
+		this._requestLockTimeouts = {};
+		this._releaseLockTimeouts = {};
 		this._responseEventEmitter = new EventEmitter();
 		this._onPrivateMessageFn = this._onPrivateMessage.bind( this );
 		this._localTopic = this._getPrivateTopic( this._options.serverName );
 		this._options.messageConnector.subscribe( this._localTopic, this._onPrivateMessageFn );
 	}
 
+	/**
+	 * Request a lock, if the leader ( whether local or distributed ) has the lock availble
+	 * it will invoke the callback with true, otherwise false.
+	 *
+	 * @param  {String}   name     the lock name that is desired
+	 * @param  {Function} callback the callback to be told if the lock has been reserved succesfully
```

Comment:
`@private & @returns` flag. I'm using [DockBlockr](https://packagecontrol.io/packages/DocBlockr) for Sublime to do most of this automatically